### DISCUSSION TITLE
DietPi Dashboard: View screenshot of active displays

### DISCUSSION THREAD

**Discussion by rmscode** [MichaIng/DietPi#7040] [Q&A]
It would be pretty neat to be able to view a screenshot of whatever is on the displays attached to the Pi. Something that updates maybe every 30 seconds or so. Doesn't have to be a live feed. I think this would be particularly useful for people who have several DietPi deployments being used for digital signage.

**Comment by MichaIng** [comment]
You mean outside of a desktop session? `fbcat` seems to do that:
```sh
apt install fbcat
fbcat /dev/fb0 > screenshot.ppm
```
In case needed, there are online converters for this PPM image format.

**Reply by MichaIng** [reply]
Ah sorry, you mean this for the DietPi-Dashboard. `fbcat` could be theoretically used for this as well to produce the screenshot, of course. But it is probably not trivial to detect what is actually shown on the display: plain console, X, or some DRM application. I am not sure whether there is any way to take a screenshot when e.g. Amiberry or something similar use KMS/DRM to display their GUI.

In every case, this needs to go to the Dashboard repo: https://github.com/ravenclaw900/DietPi-Dashboard/issues

**Comment by rmscode** [comment]
> Ah sorry, you mean this for the DietPi-Dashboard. `fbcat` could be theoretically used for this as well to produce the screenshot, of course. But it is probably not trivial to detect what is actually shown on the display: plain console, X, or some DRM application. I am not sure whether there is any way to take a screenshot when e.g. Amiberry or something similar use KMS/DRM to display their GUI.
> 
> In every case, this needs to go to the Dashboard repo: https://github.com/ravenclaw900/DietPi-Dashboard/issues

Thanks for the speedy reply. I forgot this was maintained by someone else!
